n2survey/lime/survey.py

```diff
@@ -159,6 +160,78 @@ def read_responses(self, responses_file: str) -> None:
+        if len(question_types) > 1:
+            raise ValueError(f'Question {question} has multiple types {question_types}.')
+        else:
+            question_type = question_types[0]
+
+        # get actual question response
+        if question_type == 'single-choice':
+            # ASSUME: question_responses consists only one column
+            #         in result question_group also consists of only one item
+            # create a DataFrame because questions_response is only a Series
+            response = pd.DataFrame()
+            response[question] = question_responses[question_responses.columns[0]]
+        elif question_type == 'multiple-choice':
+            # ASSUME: question response consists of multiple columns with
+            #         'Y' or NaN as entries.
+            response = question_responses.notnull()
+        elif question_type == 'free':
+            # ASSUME: question_responses consists of only one column with
+            #         unlimited options of answers.
+            response = question_responses
+        elif question_type == 'array':
+            # ASSUME: question_responses consists of multiple columns with
+            #         each column has the same set of possible choices
+            response = question_responses
+        else:
+            # raise error for unimplemented question types
+            raise ValueError(f'Unkown question type {question_type}.')
+
+        # replace labels
+        if labels:
+            if question_type == 'single-choice':
+                # rename column and the column entries
+                response = response.rename(columns=dict(question_group.label))
+                for col_idx in range(len(response.columns)):
+                    response.iloc[:, col_idx] = response.iloc[:, col_idx] \
+                        .cat.rename_categories(self.get_choices(question))
+            elif question_type == 'multiple-choice':
+                # WARNING: If choice has no possible mapping, it will be skipped.
+                # rename columns only because entries are True or False
+                response = response.rename(columns=self.get_choices(question))
+            elif question_type == 'free':
+                # rename only columns because entries are not categorical
+                response = response.rename(columns=dict(question_group.label))
+            elif question_type == 'array':
+                # rename all columns and their entries
+                response = response.rename(columns=dict(question_group.label))
+                for col_idx in range(len(response.columns)):
+                    response.iloc[:, col_idx] = response.iloc[:, col_idx] \
+                        .cat.rename_categories(self.get_choices(question))
+            else:
+                # raise error for unimplemented question types
+                raise ValueError(f'Unkown question type {question_type}.')
```

Comment:
1. Does it really matter because `self.get_choices(question)` returns always the same anyway? I just switched to DataFrame.replace.
2. I do not understand what you want to check with `question_group["column"].choices is not None`. I understand that you want to keep the code as simple and short as possible but I also think shortening the code too much makes it harder to maintain for others. So, switching to the if statement above is in my opinion more confusing than iterating over all question types we support. Also, for a single if statement we get unexpected behaviour for unsupported question types or not?